apps/condo/domains/meter/schema/RegisterMetersReadingsService.js

```diff
@@ -0,0 +1,260 @@
+                'commissioningDate: String,' +
+                'sealingDate: String,' +
+                'controlReadingsDate: String' +
+                '}',
+        },
+        {
+            access: true,
+            type: 'input RegisterMetersReadingsReadingAddressMetaInput { unitType: String!, unitName: String!, globalId: String }',
+        },
+        {
+            access: true,
+            type: 'input RegisterMetersReadingsReadingInput {' +
+                'address: String!,' +
+                'addressMeta: RegisterMetersReadingsReadingAddressMetaInput!,' +
+                'accountNumber: String!,' +
+                'meterNumber: String!,' +
+                'meterResource: MeterResourceWhereUniqueInput!,' +
+                'date: String!,' +
+                'value1: String!,' +
+                'value2: String,' +
+                'value3: String,' +
+                'value4: String,' +
+                'meterMeta: RegisterMetersReadingsMeterMetaInput' +
+                '}',
+        },
+        {
+            access: true,
+            type: 'input RegisterMetersReadingsInput { dv: Int!, sender: SenderFieldInput!, organization: OrganizationWhereUniqueInput!, readings: [RegisterMetersReadingsReadingInput!]! }',
+        },
+    ],
+
+    mutations: [
+        {
+            access: access.canRegisterMetersReadings,
+            schema: 'registerMetersReadings(data: RegisterMetersReadingsInput!): [MeterReading]',
+            resolver: async (parent, /**{ data: RegisterMetersReadingsInput }*/args, context) => {
+                const { data: { dv, sender, organization, readings } } = args
+
+                if (readings.length > READINGS_LIMIT) {
+                    throw new GQLError(ERRORS.TOO_MUCH_READINGS(readings.length), context)
+                }
+
+                /** @type Organization */
+                const organizationData = await getById('Organization', organization.id)
+
+                if (!organizationData) {
+                    throw new GQLError(ERRORS.ORGANIZATION_NOT_FOUND, context)
+                }
+
+                const propertyResolver = new PropertyResolver({ context })
+                propertyResolver.tin = organizationData.tin
+
+                const resolvedAddresses = await propertyResolver.normalizeAddresses(readings.reduce((res, reading, index) => ({
+                    ...res,
+                    [reading.address]: {
+                        address: reading.address,
+                        addressMeta: reading.addressMeta,
+                    },
+                }), {}))
+
+                const propertyOrConditions = []
+                for (const reading of readings) {
+                    const addressKey = get(resolvedAddresses, [reading.address, 'addressResolve', 'propertyAddress', 'addressKey'])
+                    propertyOrConditions.push({ ...addressKey ? { OR: [{ address_i: reading.address }, { addressKey }] } : { address_i: reading.address } })
```

Comment:
No, that's not a deal. Use `addressKey`!

Reply:
Why did we migrate to addressKey if after 2 years we can't use it?

Reply:
Fixed